modules/tensor_mechanics/src/materials/ComputeMultiPlasticityStress.C

```diff
@@ -1375,9 +1375,18 @@ ComputeMultiPlasticityStress::consistentTangentOperator(const RankTwoTensor & st
   if (num_currently_active > 0)
   {
     // invert zzz, in place.  if num_currently_active = 0 then zzz is not needed.
-    const int ierr = MatrixTools::inverse(zzz, num_currently_active);
-    if (ierr != 0)
-      return E_ijkl; // in the very rare case of zzz being singular, just return the "elastic" tangent operator
+    try
+    {
+      MatrixTools::inverse(zzz, num_currently_active);
+    }
+    catch(const MooseException & e)
```

Comment:
If the exception is for being _singular_ then it should be `MatrixTools::SingularException`